### PR TITLE
feat(attach): avoid creation of volumeattachment object

### PR DESCRIPTION
**Pull request by pawanpraka1**
fixes :- https://github.com/openebs/zfs-localpv/issues/79

k8s is very slow in attaching the volumes when dealing with the 
large number of volume attachment object.

(k8s issue https://github.com/kubernetes/kubernetes/issues/84169)

The volumeattachment is not required for ZFSPV, so avoid creation
of attachment object, also removed the csi-attacher container as
this is also not needed as it acts on volumeattachment object.

k8s is very slow in attaching the volumes when dealing with the 
large number of volume attachment object : 

k8s issue https://github.com/kubernetes/kubernetes/issues/84169).

Volumeattachment is a CR created just to tell the watcher of it
which is csi-attacher, that it has to call the Controller Publish/Unpublish grpc.
Which does all the tasks to attach the volumes to a node for example call to the 
DigitalOcean Block Storage API service to attach a created volume to a specified node.
Since for ZFSPV, volume is already present locally, nothing needs to done in Controller
Publish/Unpublish, so it is good to remove them.

so avoiding creation of attachment object in this change, also removed the csi-attacher
container as this is also not needed as it acts on volumeattachment object.

Removed csi-cluster-driver-registrar container also as it is deprecated and not needed anymore.

We are using csidriver beta CRDs so minimum k8s version required is 1.14+.

Signed-off-by: Pawan <pawan@mayadata.io>

### CODE REVIEW

File: upgrade/cleanup.sh [hunk 1]
```diff
@@ -2,27 +2,53 @@
...
-# remove the finalizer from the old CR
-sed -i "/zfs.openebs.io\/finalizer/d" volumes.yaml
-kubectl apply -f volumes.yaml
+echo "Fetching ZFS Volumes"
+numVol=`kubectl get zfsvolumes.openebs.io --no-headers -n $ZFSPV_NAMESPACE | wc -l`
 
-# delete the old CR
-kubectl delete -f volumes.yaml
+if [ $numVol -gt 0 ]; then
+	echo "Cleaning the ZFS Volumes(`expr $numVol - 1`)"
```
Comment (w3aman) [inline upgrade/cleanup.sh:16]: why $numVol - 1  ??

Comment (pawanpraka1) [reply upgrade/cleanup.sh]: earier I was not using --no-headers, which will give one extra count, so added -1 here. Now added  --no-headers but forgot to fix this message.  Thanks @w3aman .fixed it

File: deploy/yamls/zfs-driver.yaml [hunk 2]
```diff
@@ -543,7 +559,7 @@ spec:
           image: quay.io/k8scsi/snapshot-controller:v2.0.1
           args:
             - "--v=5"
-            - "--leader-election=false"
+            - "--leader-election=true"
```
Comment (kmova) [inline deploy/yamls/zfs-driver.yaml:562]: why is this required?

Comment (pawanpraka1) [reply deploy/yamls/zfs-driver.yaml:562]: needed if HA customer is using HA.
